src/Document.php

```diff
@@ -707,6 +707,11 @@ public function canViewFile(array $options = [])
         $itemtype = $options['itemtype'] ?? null;
         $items_id = $options['items_id'] ?? null;
 
+        // Validate itemtype, as it might come directly from $_GET
+        if ($itemtype !== null && !Toolbox::isCommonDBTM($itemtype)) {
```

Comment:
Is any class that inherits from `CommonGLPI` and not `CommonDBTM` possible here?

Reply:
I think `itemtype` implies to also have some kind of `items_id`, which would indicate a `CommonDBTM`.
As far as a know, we can't link documents to any `CommonGLPI` existing class.